### PR TITLE
Add board definition for Sparkfun Pro Micro RP2040

### PR DESCRIPTION
**Pull request by spe2**
Adds appropriate board definition files for the Pro Micro RP2040 and adds it to the build.yml file.

### CODE REVIEW

File: ports/raspberrypi/boards/sparkfun_pro_micro_rp2040/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#define MICROPY_HW_BOARD_NAME "Sparkfun Pro Micro RP2040"
```
Comment (tannewt) [inline ports/raspberrypi/boards/sparkfun_pro_micro_rp2040/mpconfigboard.h:1]:
```suggestion
#define MICROPY_HW_BOARD_NAME "SparkFun Pro Micro RP2040"
```

File: ports/raspberrypi/boards/sparkfun_pro_micro_rp2040/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,9 @@
+USB_VID = 0x1B4F
+USB_PID = 0x0026
+USB_PRODUCT = "Pro Micro RP2040"
+USB_MANUFACTURER = "Sparkfun"
```
Comment (tannewt) [inline ports/raspberrypi/boards/sparkfun_pro_micro_rp2040/mpconfigboard.mk:4]:
```suggestion
USB_MANUFACTURER = "SparkFun"
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Two minor capitalization changes. Good otherwise. Thank you!

**Comment by spe2** [comment]
@tannewt Suggested changes committed, thanks.

**Review by tannewt — APPROVED** [review]
Thanks!

**Pull request merged**